### ISSUE TITLE
The question mark in the proxy example in the doc `-P http://localhost:8080?` is causing issue

### ISSUE THREAD

**Issue by lz000** [http-party/http-server#645]
**Steps to reproduce the issue, if applicable. Include the actual command and output and/or stack trace.**

I ran `http-server /data -p 80 -P http://localhost:8080?` as suggested in the doc. When I hit http://localhost/test, my downstream server (which listening on 8080) get `/?/test`

**What did you expect to happen?**

Maybe I don't understand the purpose of the `?` at the end. Removing the `?` fix the issue. Is this a mistake in the doc or I missed anything?

**Tell us about your environment**
- **exact http-server version:** 0.12.3
- **Node version:** 12.18.3
- **Platform:** Windows 10

**Comment by elwinschmitz** [comment]
I think I had the same misunderstanding of that section in the README; As I thought it was explaining the use of the `--proxy`-option, but actually it isn't.

That section in the README tries to explain a clever 'workaround' for when you want to test a Single-Page-App which handles its own urls (like Angular/React/Vue can with their Router-components).

The use-case you where looking for (similar to mine, I think), is to run your static front-end on http://loocalhost:80 and let all requests to `/api/` for example, go to <http://backend-server.example.com/api/> (or locally to http://localhost:8080/api/)

To do that, you would be using 'just' the `--proxy`-option:
`npx http-server --port 80 --proxy http://backend.example.com/`

The `?` in the URL in the README is only there to 'pass' the rest of the URL to your local `index.html` and let your framework's Router handle it.

**Comment by thornjad** [comment]
@elwinschmitz is correct, that part of the docs is only a suggestion for a workaround for single-page apps, which http-server does not explicitly support (and won't). To use the proxy normally, you probably don't want the `?`.

**Comment by rook2pawn** [comment]
> I think I had the same misunderstanding of that section in the README; As I thought it was explaining the use of the `--proxy`-option, but actually it isn't. (See the discussion: [#338 (comment)](https://github.com/http-party/http-server/issues/338#issuecomment-338601211))
> 
> That section in the README tries to explain a clever 'workaround' for when you want to test a Single-Page-App which handles its own urls (like Angular/React/Vue can with their Router-components).
> 
> The use-case you are looking for (similar to mine, I think), is to run your static front-end on http://loocalhost:80 and let all requests to `/api/` for example, go to http://back-end.example.com/api/ (or locally to http://localhost:8080/api/)
> 
> To do that, you would be using 'just' the `--proxy`-option: `npx http-server --port 80 --proxy http://back-end.example.com/`
> 
> The `?` in the URL in the README is only there to 'pass' the rest of the URL to your local `index.html` and let your framework's Router handle it.

@elwinschmitz I have a SPA and i need this kind of setup to make my SPA work with http-server
```
http-server ./web -o -c-1 -p 8080 --proxy http://localhost:8080?
```
however I also have a backend on 5150 and want to proxy /api to 5150.. Basically I want to use both the catch-all redirect for SPA but also a proxy to the backend to another port for anything request like /api/<something>. I tried looking at the options in --proxy-options and couldn't figure out how to do this.  You said you used the proxy to pass to your BE - I assume you weren't needing a catchall redirect for SPA on your FE?

**Comment by elwinschmitz** [comment]
@rook2pawn I only use it for the 'fake' backend indeed. 

But I think you can run 2 instances of `http-server` next to each other, with an ` & ` between the commands. 
Or with a tool like `npm-run-all`, to run 2 tasks in parallel.

**Comment by rook2pawn** [comment]
@elwinschmitz I was very happy with the solution i am using now and i posted it [here](https://github.com/choojs/choo/issues/712#issuecomment-939372462). Essentially I set up a vanilla https server, and then proxy everything that's starts with `/api/` to the BE server, and everything else proxies to the `http-server` for my SPA FE server. It works well and wanted to document that somewhere. What's neat is that I can now have interesting features like rewriting, load balance on my BE and even things that could be done on the BE that are better done in the proxy like blacklist and simple ddos protection. I was really happy that I didn't hit any major snags in my setup.

**Comment by elwinschmitz** [comment]
Aha. That's sounds neat. 
I am using this in my local development setup only, so any 'real usage'/load/security concerns are 'out of scope' for me. 😉

**Comment by rook2pawn** [comment]
![diagram](https://i.postimg.cc/8PscGFBf/proxydiagram.png) is the basic setup that i use - what's nice about this approach is that you have a clean separation between your SPA and your BE (repo, hosting/instances) and you can even put all sorts of things at your proxy level (administrative redirects, prevent spam attacks) so your logic on your SPA/BE/proxy are all cleanly split into the domains they are best suited for.
